assets/src/edit-story/components/panels/link.js

```diff
@@ -31,83 +31,66 @@ import { __ } from '@wordpress/i18n';
  */
-import { Input as BaseInput, Row } from '../form';
+import { TextInput, Media, Row } from '../form';
 import { createLink } from '../link';
-import { ReactComponent as Close } from '../../icons/close_icon.svg';
 import { SimplePanel } from './panel';
 import getCommonValue from './utils/getCommonValue';
 
-const Note = styled.span`
-  color: ${({ theme }) => theme.colors.mg.v1};
-  font-size: 11px;
-  line-height: 16px;
-  margin-bottom: 6px;
-`;
-
-/** TODO(@wassgha): Replace with text input component once done */
-const Input = styled(BaseInput)`
-  width: 100%;
-  border: none !important;
-  margin: 0;
-  border-radius: 4px !important;
-`;
-
-/** TODO(@wassgha): Replace with text input component once done */
-const InputContainer = styled.div`
-  position: relative;
-  width: 100%;
-  background: ${({ theme, disabled }) =>
-    disabled ? theme.colors.fg.v3 : theme.colors.fg.v1};
-  color: ${({ theme }) => theme.colors.mg.v1};
-  font-family: ${({ theme }) => theme.fonts.body2.family};
-  font-size: ${({ theme }) => theme.fonts.body2.size};
-  border: 1px solid ${({ theme }) => theme.colors.fg.v3} !important;
-  border-radius: 4px !important;
-  box-sizing: border-box;
-  display: flex;
-  align-items: center;
+const BoxedTextInput = styled(TextInput)`
+  padding: 6px 6px;
+  border-radius: 4px;
 `;
 
-const BrandIcon = styled.img`
-  width: 64px;
-  height: 64px;
-  border-radius: 50%;
-  background-color: ${({ theme }) => theme.colors.fg.v3};
-  border: none;
-  margin-right: 8px;
+const ExpandedTextInput = styled(BoxedTextInput)`
+  flex-grow: 1;
 `;
 
-const DeleteButton = styled.button`
-  position: absolute;
-  right: 8px;
-  appearance: none;
-  background: ${({ theme }) => rgba(theme.colors.fg.v0, 0.54)};
-  width: 16px;
-  height: 16px;
-  border: none;
-  padding: 0px;
-  border-radius: 50%;
-  display: flex;
-  justify-content: center;
-  align-items: center;
-`;
-
-const CloseIcon = styled(Close)`
-  width: 12px;
-  height: 12px;
+const Note = styled.span`
+  color: ${({ theme }) => rgba(theme.colors.fg.v1, 0.54)};
+  font-family: ${({ theme }) => theme.fonts.body1.family};
+  font-size: 12px;
+  line-height: 16px;
 `;
 
 function LinkPanel({ selectedElements, onSetProperties }) {
   const link = getCommonValue(selectedElements, 'link') || null;
   const isFill = getCommonValue(selectedElements, 'isFill');
+
   const [state, setState] = useState({ link: createLink() });
   useEffect(() => {
-    setState({ link });
+    setState({ ...link });
   }, [link]);
-  const handleSubmit = (evt) => {
-    onSetProperties(state);
-    evt.preventDefault();
-  };
+
+  const handleChange = useCallback(
+    (property) => (value) =>
+      setState((originalState) => ({
+        ...originalState,
+        [property]: value,
+      })),
+    [setState]
+  );
+  const handleChangeIcon = useCallback(
+    (image) => {
+      const icon =
+        image.sizes && image.sizes.medium ? image.sizes.medium.url : image.url;
```

Comment:
I think that `image?.sizes?.medium ? ... : ....` will work in the same way.

Reply:
Yup